modules/c++/types/include/types/Range.h

```diff
@@ -89,28 +89,52 @@ struct Range
     }
 
     /*!
-     * \param startElementToTest The start element
-     * \param numElementsToTest The total number of elements to check
+     * \param rhs Range to compare with
      *
-     * \returns The number of shared elements
+     * \return True if the ranges overlap, false otherwise
      */
-    size_t getNumSharedElements(size_t startElementToTest,
-                                size_t numElementsToTest) const
+    bool overlaps(const types::Range& rhs) const
     {
-        const size_t endElementToTest =
-                startElementToTest + numElementsToTest;
+        return (endElement() > rhs.mStartElement &&
```

Comment:
Should this by >= ? If your end element equals my start element, I would consider that to be overlapping a single element.

Reply:
End element is exclusive...  So if lhs `[start, end)` is `[100, 200)` and rhs is `[200, 300)`, they don't overlap

Reply:
Ok, then I agree this logic is correct. I think this is good to merge.